src/bp-activity/bp-activity-notifications.php

```diff
@@ -476,3 +476,45 @@ function bp_activity_screen_notification_settings() {
 	<?php
 }
 add_action( 'bp_notification_settings', 'bp_activity_screen_notification_settings', 1 );
+
+/**
+ * Mark notifications as read when a user visits an single post.
+ *
+ * @since BuddyBoss 1.5.2
+ */
+function bp_activity_remove_screen_notifications_single_post() {
+
+	if (
+		! is_single()
+		|| (
+			( ! isset( $_GET['rid'] ) || empty( $_GET['rid'] ) )
```

Comment:
@KartikSuthar From now onward we need to start using some WordPress best practice.  
LIKe For REQUEST DATA LIKE $_POST, $_GET, and $_REQUEST, We should filter it before use it  
Ref: https://www.w3schools.com/php/func_filter_input.asp

Reply:
Thanks, @chetansatasiya, I will follow always in the future.